CONTRIBUTING.md

```diff
@@ -6,29 +6,300 @@ This page is intended for contributors to the registry and packages.
 * Zero or more ingest pipelines
 * An Elastic Agent config template
 
-## How to build a package
+### Migration from Beats
+
+A defined importing procedure used to transform both, Filebeat and Metricbeat modules, related to
+the same observed product, into a single integration. The integration contains extracted dataset configuration of beat
+modules, hence no modules are required to exist anymore.
+
+## Package structure
+
+### Elements
+
+Link: https://github.com/elastic/package-registry/blob/master/ASSETS.md
+
+### Reference packages
+
+The following packages can be considered as reference points for all integrations.
+
+#### Integration: reference
+
+Link: https://github.com/elastic/package-registry/tree/master/dev/packages/example/reference-1.0.0
+
+The directory contains mandatory manifest files defining the integration and its datasets. All manifests have fields
+annotated with comments to better understand their goals.
+
+_Keep in mind that this package doesn't contain all file resources (images, screenshots, icons) referenced in manifests.
+Let's assume that they're also there._
+
+#### Integration: mysql
+
+Link: https://github.com/mtojek/package-registry/tree/package-mysql-0.0.2/dev/packages/alpha/mysql-0.0.2
+
+The MySQL integration was the first integration built using the [import-beats](https://github.com/elastic/package-registry/tree/master/dev/import-beats) script.
+The script imported filesets and metricsets from both MySQL modules, and converted them to a package.
+
+The MySQL integration contains all parts that should be present (or are required) in the integration package.
+
+After using the _import-beats_ script, the integration has been manually adjusted and extended with dedicated docs.
+
+## Create a new integration
+
+This section describes steps required to build a new integration. If you plan to prepare the integration
+with a product unsupported by [Beats](https://github.com/elastic/beats), feel free to skip the section about importing
+existing modules.
+
+### Import from existing modules
+
+The import procedure heavily uses on the _import-beats_ script. If you are interested how does it work internally,
+feel free to review the script's [README](https://github.com/elastic/package-registry/blob/master/dev/import-beats/README.md).
+
+1. Focus on the particular product (e.g. MySQL, ActiveMQ) you would like to integrate with.
+2. Prepare the developer environment:
+    1. Clone/refresh the following repositories:
+        * https://github.com/elastic/beats
+        * https://github.com/elastic/ecs
+        * https://github.com/elastic/eui
+        * https://github.com/elastic/kibana
+
+       Make sure you don't have any manual changes applied as they will reflect on the integration.
+    2. Clone/refresh the Elastic Package Registry (EPR) to always use the latest version of the script:
+        * https://github.com/elastic/package-registry
+    3. Make sure you've the `mage` tool installed.
```

Comment:
nit: how about adding the command line for this?

Reply:
Fixed. Added a one-liner for mage.